modules/core/settings/actions.js

```diff
@@ -88,3 +96,74 @@ export function getLanguage(event) {
         }
     }
 }
+
+export function toggleSwarm(event) {
+    return (dispatch, getState) => {
+        if ([SwarmState.Enabled, SwarmState.Enabling].indexOf(getState().settings.swarmState) > -1) {
+            dispatch({ type: '[MAIN]:SWARM:STOP' });
+
+            try {
+                swarmNode.on('stopping', () => {
+                    swarmLog.info('Stopping Swarm');
+                    dispatch({ type: '[MAIN]:SWARM:DISABLING' });
+                });
+
+                swarmNode.on('stopped', () => {
+                    swarmLog.info('Swarm stopped');
+                    dispatch({ type: '[MAIN]:SWARM:DISABLED' });
+                    dispatch(resetMenu());
+                });
+
+                swarmNode.stop();
+
+            } catch (error) {
+                dispatch({ type: '[MAIN]:SWARM:FAILURE', error });
+                swarmLog.error(error);
+            }
+
+        } else {
+            dispatch({ type: '[MAIN]:SWARM:START' });
+
+            try {
+                swarmNode.on('starting', () => {
+                    swarmLog.info('Starting Swarm');
+                    dispatch({ type: '[MAIN]:SWARM:ENABLING' });
+                });
+
+                swarmNode.on('downloadProgress', (progress) => {
+                    swarmLog.info(progress);
+                });
+
+                swarmNode.on('started', () => {
+                    swarmLog.info('Swarm started');
+                    dispatch({ type: '[MAIN]:SWARM:ENABLED' });
+                    dispatch(resetMenu());
+                });
+
+                swarmNode.init();
+
+            } catch (error) {
+                dispatch({ type: '[MAIN]:SWARM:FAILURE', error });
+                swarmLog.error(error);
+            }
+        }
+    }
+}
+
+export function toggleSwarmOnStart(event) {
+    return (dispatch, getState) => {
+        if (getState().settings.swarmEnableOnStart) {
+            Settings.enableSwarmOnStart = false;
+            dispatch({ type: '[MAIN]:SWARM:DISABLE_ON_START' });
+        } else {
+            Settings.enableSwarmOnStart = true;
+            dispatch({ type: '[MAIN]:SWARM:ENABLE_ON_START' });
+        }
+    }
+}
+
+export function setSwarmEnableOnStart() {
+    return (dispatch, getState) => {
+        dispatch({ type: '[MAIN]:SWARM:ENABLE_ON_START' });
+    }
```

Comment:
Whenever you don't need to perform any side-effects, you can just use a synchronous action creator. i.e.
```
export function setSwarmEnableOnStart() {
   return { type: '[MAIN]:SWARM:ENABLE_ON_START' };
}
```